include/net/kfi/kfi_provider.h

```diff
@@ -33,18 +33,10 @@
 #ifndef _KFI_PROVIDER_H_
 #define _KFI_PROVIDER_H_
 
-#include <linux/kernel.h>
-#include <linux/byteorder/generic.h>
-#include <linux/semaphore.h>
-#include <linux/list.h>
-#include <linux/string.h>
```

Comment:
I didn't check the code, but note that headers should always be included directly if they are needed, and not indirectly via a separate header file.